src/homework4/functions.py

```diff
@@ -0,0 +1,45 @@
+"""Реализовать функцию get_ranges которая получает на вход непустой список
+неповторяющихся целых чисел, отсортированных по возрастанию, которая этот
+список “сворачивает”
+ get_ranges([0, 1, 2, 3, 4, 7, 8, 10]) // "0-4,7-8,10"
+ get_ranges([4,7,10]) // "4,7,10"
+ get_ranges([2, 3, 8, 9]) // "2-3,8-9"
+"""
+from functools import reduce
+
+
+# first solution, works only with positive numbers
+def get_ranges(int_list):
+    return ",".join(reduce(gluing, int_list, []))
+
+
+def gluing(res_lst, num):
+    if res_lst:
+        last = res_lst[-1].rpartition("-")[-1]
```

Comment:
I've studied rpartition function 🧪

Reply:
here it works like split, I unnecessarily complicated